optimum/exporters/onnx/config.py

```diff
@@ -211,3 +212,88 @@ def generate_dummy_inputs_for_validation(self, reference_model_inputs: Mapping[s
             reference_model_inputs["encoder_hidden_states"] = reference_model_inputs.pop("encoder_outputs")[0]
 
         return reference_model_inputs
+
+
+class EncoderDecoderOnnxConfig(OnnxSeq2SeqConfigWithPast):
+    DUMMY_INPUT_GENERATOR_CLASSES = (
+        DummyTextInputGenerator,
+        DummySeq2SeqDecoderTextInputGenerator,
+    )
+
+    def __init__(
+        self,
+        config: "PretrainedConfig",
+        task: str = "default",
+        patching_specs: Optional[List["PatchingSpec"]] = None,
+        use_past: bool = False,
+        use_past_in_inputs: Optional[bool] = None,
+        use_present_in_outputs: Optional[bool] = None,
+        behavior: ConfigBehavior = ConfigBehavior.MONOLITH,
+    ):
+        super().__init__(
+            config,
+            task=task,
+            patching_specs=patching_specs,
+            use_past=use_past,
+            use_past_in_inputs=use_past_in_inputs,
+            use_present_in_outputs=use_present_in_outputs,
+            behavior=behavior,
+        )
+
+        from ..tasks import TasksManager
+
+        if self._behavior is not ConfigBehavior.DECODER:
+            # retrieve the encoder config
+            encoder_onnx_config_constructor = TasksManager.get_exporter_config_constructor(
+                exporter="onnx", task="default", model_type=config.encoder.model_type
+            )
+            self._encoder_onnx_config = encoder_onnx_config_constructor(config.encoder)
+            self._normalized_config.ENCODER_NORMALIZED_CONFIG_CLASS = self._encoder_onnx_config._normalized_config
+
+        if self._behavior is not ConfigBehavior.ENCODER:
+            # retrieve the decoder config
+            decoder_onnx_config_constructor = TasksManager.get_exporter_config_constructor(
+                exporter="onnx", task="default", model_type=config.decoder.model_type
+            )
+            self._decoder_onnx_config = decoder_onnx_config_constructor(config.decoder, use_past=use_past)
+
+            self._normalized_config.DECODER_NORMALIZED_CONFIG_CLASS = self._decoder_onnx_config._normalized_config
+
+            if isinstance(self._decoder_onnx_config, OnnxSeq2SeqConfigWithPast):
+                self._past_key_values_generator = (
+                    DummySeq2SeqDecoderTextInputGenerator,
+                    DummySeq2SeqPastKeyValuesGenerator,
+                )
+            else:
+                self._past_key_values_generator = (
+                    DummySeq2SeqDecoderTextInputGenerator,
+                    DummyPastKeyValuesGenerator,
+                )
+
```

Comment:
I would add the nromalized config creation hear instead of setting attributes in the if branches. I would also add a type check on it.

Reply:
Is there a reason to set the values in constructor than setting attribute? Plus why would type check be necessary?